ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/resources/streaming/PullQueryPublisher.java

```diff
@@ -141,27 +144,30 @@ public synchronized void subscribe(final Subscriber<Collection<StreamedRow>> sub
       decrementer.decrementAtMostOnce();
 
       if (result == null) {
-        pullQueryMetrics.ifPresent(m -> recordMetrics(m, Optional.empty()));
+        pullQueryMetrics.ifPresent(this::recordErrorMetrics);
       }
       throw t;
     }
   }
 
   private void recordMetrics(
-      final PullQueryExecutorMetrics metrics, final Optional<PullQueryResult> result) {
-    final PullSourceType sourceType = result.map(
-        PullQueryResult::getSourceType).orElse(PullSourceType.UNKNOWN);
-    final PullPhysicalPlanType planType = result.map(
-        PullQueryResult::getPlanType).orElse(PullPhysicalPlanType.UNKNOWN);
-    final RoutingNodeType routingNodeType = result.map(
-        PullQueryResult::getRoutingNodeType).orElse(RoutingNodeType.UNKNOWN);
+      final PullQueryExecutorMetrics metrics, final PullQueryResult result) {
```

Comment:
I noticed that we're not recording the response size here... Is that intentional? (cc @cprasad1  )

Reply:
Yes, it is.   This is the web socket interface and we couldn't easily figure out how to report response size, unfortunately.  So we don't have complete metrics for this interface, though it's not used outsize of the UI.